proposals/NNNN-android-transport-overhaul.md

```diff
@@ -0,0 +1,490 @@
+			}
+		}
+	}
+}
+
+```
+
+#### Packet Queues
+Packet queues will be a per transport per app item. Each bound app will have a queue per their registered transport for all sessions to operate.
+
+```java
+HashMap<TransportType, PacketWriteTaskBlockingQueue> queues;
+
+public boolean handleIncommingClientMessage(final Bundle receivedBundle){
+	int flags = receivedBundle.getInt(TransportConstants.BYTES_TO_SEND_FLAGS, TransportConstants.BYTES_TO_SEND_FLAG_NONE);
+	TransportType transportType = TransportType.valueForString(receivedBundle.getString(TransportConstants.TRANSPORT_FOR_PACKET));
+
+	if(flags!=TransportConstants.BYTES_TO_SEND_FLAG_NONE){
+		byte[] packet = receivedBundle.getByteArray(TransportConstants.BYTES_TO_SEND_EXTRA_NAME); 
+		if(flags == TransportConstants.BYTES_TO_SEND_FLAG_LARGE_PACKET_START){
+			this.priorityForBuffingMessage = receivedBundle.getInt(TransportConstants.PACKET_PRIORITY_COEFFICIENT,0);
+		}
+		handleMessage(flags, packet, transportType);
+	}else{
+		//Add the write task on the stack
+		PacketWriteTaskBlockingQueue queue = queues.get(transportType);
+		if(queue == null){	//TODO check to see if there is any better place to put this
+			queue = new PacketWriteTaskBlockingQueue();
+			queues.put(transportType,queue);
+		}
+		if(queue!=null){
+			queue.add(new PacketWriteTask(receivedBundle));
+			PacketWriteTaskMaster packetWriteTaskMaster = packetWriteTaskMasterMap.get(transportType);
+			if(packetWriteTaskMaster!=null){
+				packetWriteTaskMaster.alert();
+			}
+		}
+	}
+	return true;
+}
+
+```
+
+#### SdlPacket Update
+
+The `SdlPacket` class will now include a `TransportRecord` for the local library. This lets the router service know which transport to send a packet. Also, when the router service receives a packet from the core module and relays it to the bound client app, the client knows where the packet came from.
+
+```java
+public class TransportRecord implements Parcelable{
+
+    TransportType type;
+    String address;
+
+    public TransportRecord(TransportType transportType, String address){
+        this.type = transportType;
+        this.address = address;
+    }
+
+    ....    
+}
+```
+
+
+#### Secondary Transport Registration
+
+If the router service notices a Register Transport control frame ACK is received it will look for the corresponding session, register that transport for that session on that registered app, and forward the packet to the client.
+
+### Breaking change phase
+
+
+#### Remove WiProProtocol, SdlConnection, and Listener
+
+All of these will no longer be necessary and should be removed.
+
+#### Move SdlSession Class
+
+With `SdlConnection` and listener removed, the package `com.smartdevicelink.SdlConnection` will no longer be valid and keeping `SdlSession` there will be orphaned. The class will be moved to `com.smartdevicelink.proxy`.
+
+#### Misc Cleanup of Deprecated items
+
+It will be required to remove all deprecated methods previously by this proposal's implementation.
```

Comment:
I think this sentence needs to be updated.  Maybe "It will be required to remove all deprecated methods previously mentioned in this proposal that are replaced by this proposal's implementation." ?